src/main/scala/strawman/collection/IterableFactory.scala

```diff
@@ -0,0 +1,35 @@
+package strawman
+package collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Int
+
+/** Base trait for instances that can construct a collection from an iterable */
+trait FromIterable[+C[X] <: Iterable[X]] {
+  def fromIterable[B](it: Iterable[B]): C[B]
+}
+
+/** Base trait for builder factories */
+trait CanBuild[-A, +Repr] {
```

Comment:
Okay, but in the former case we'd need only
`implicit def canBuild[A] = () => newBuilder[A, C[A]]` instead of the four lines we have now.